src/doc/book/structs.md

```diff
@@ -117,6 +117,28 @@ fn main() {
 }
 ```
 
+We can initializing a data structure (struct, enum, union) with named fields, by writing `fieldname` as a shorthand for `fieldname: fieldname`. This allows a compact syntax for initialization, with less duplication:
```

Comment:
This sentence doesn't read write to me at all. I would replace all of `a data structure (struct, enum, union) with named fields` with `named fields`.

I would also add a sentence saying that a variable of the name of the field also has to be in scope.